### PR TITLE
feat(ModalContainer): require label

### PR DESCRIPTION
**Pull request by nataliadelmar**
All modal containers should have an aria-label or aria-labelledby that labels them.

### CODE REVIEW

File: src/components/OverlayAlert/OverlayAlert.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const OverlayAlert: FC<Props> = (props: Props) => {
         className={classnames(STYLE.modalContainer)}
         round={75}
         color={modalColor}
-        aria-label={ariaLabel}
+        aria-label={ariaLabel || title}
```
Comment (lismith2-cisco) [inline src/components/OverlayAlert/OverlayAlert.tsx:64]: this seems unnecessary, as if there's a title, there's an aria-labelledby, but presumably it makes tsc happy?

File: src/components/ModalContainer/ModalContainer.unit.test.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ describe('<ModalContainer />', () => {
     it('should match snapshot', () => {
       expect.assertions(1);
 
-      const container = mount(<ModalContainer />);
+      const container = mount(<ModalContainer aria-label="Some label" />);
```
Comment (lismith2-cisco) [inline src/components/ModalContainer/ModalContainer.unit.test.tsx:15]: Should add explicit tests for aria-label and aria-labelledby

**Review by lismith2-cisco — CHANGES_REQUESTED** [review]

File: src/components/ToastNotification/ToastNotification.unit.test.tsx [hunk 3]
```diff
@@ -167,7 +186,7 @@ describe('<ToastNotification />', () => {
       const ariaLabel = 'test';
 
       const wrapper = await mountAndWait(
-        <ToastNotification ariaLabel={ariaLabel} content={exampleContent} />
+        <ToastNotification aria-label={ariaLabel} content={exampleContent} />
```
[the remark this answers is not in the document]
Comment (nataliadelmar) [reply src/components/ToastNotification/ToastNotification.unit.test.tsx:189]: Added :)

File: src/components/ToastNotification/ToastNotification.types.ts [hunk 4]
```diff
@@ -6,11 +6,12 @@ import { IconProps } from '../Icon';
 import { BadgeProps } from '../Badge';
 import { ButtonPillProps } from '../ButtonPill';
 import { ButtonCircleProps } from '../ButtonCircle';
+import { AriaLabelRequired } from '../../utils/a11y';
 
 export type SupportedLeadingVisuals = AvatarProps | IconProps | BadgeProps;
 export type SupportedButtonGroup = ButtonPillProps | ButtonCircleProps;
 
-export interface Props {
+export type Props = AriaLabelRequired & {
```
[the remark this answers is not in the document]
Comment (nataliadelmar) [reply src/components/ToastNotification/ToastNotification.types.ts:14]: yeah; I can imagine a case in which it is useful: either aria-label or aria-labelledby are required; that is my intention

File: src/components/ToastNotification/ToastNotification.stories.tsx [hunk 5]
```diff
@@ -20,6 +20,9 @@ export default {
       page: DocumentationPage(Documentation, StyleDocs),
     },
   },
+  args: {
+    'aria-label': 'Some label',
```
[the remark this answers is not in the document]
Comment (nataliadelmar) [reply src/components/ToastNotification/ToastNotification.stories.tsx:24]: it is optional, right? I'm requiring either aria-label or aria-labelledby

File: src/components/OverlayAlert/OverlayAlert.tsx (hunk 1 above)
Comment (nataliadelmar) [reply src/components/OverlayAlert/OverlayAlert.tsx:64]: Oh I hadn't noticed that Charles added the title to aria-label the ModalContainer if a title is passed and no aria-label is specified; all good then

Comment (nataliadelmar) [reply src/components/OverlayAlert/OverlayAlert.tsx:64]: I'll delete this change

File: src/components/ModalContainer/ModalContainer.unit.test.tsx (hunk 2 above)
Comment (nataliadelmar) [reply src/components/ModalContainer/ModalContainer.unit.test.tsx:15]: Added :)

File: src/components/ToastNotification/ToastNotification.stories.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (lismith2-cisco) [reply src/components/ToastNotification/ToastNotification.stories.tsx:24]: sorry I didn't look properly cos I was hurrying! I was thinking this was the table of parameters for some reason!

**Review by lismith2-cisco — APPROVED** [review]

**Pull request merged**